app/utils/orgs.js

```diff
@@ -68,4 +71,55 @@ const decryptOrgData = (orgKey, data) => {
   return tokenCrypt.decrypt(data, orgKey);
 };
 
-module.exports = { getOrg, verifyAdminOrgKey, encryptOrgData, decryptOrgData};
+var getKmsKeyForOrg = async({ org, metroConf })=>{
+  if(!org?._id){
+    throw new Error('genKmsKeyForOrg requires an org as input');
+  }
+  if(!metroConf){
+    throw new Error('genKmsKeyForOrg requires metroConf in input');
+  }
+  var crn = _.get(org, 'kms.crn');
+  if(crn){
+    // if already has a key, then returns it
+    return crn;
+  }
+  var name = `s3_org_encryption_key_${org._id}`;
+  crn = await genKmsKey({ name, metroConf });
+  var search = {
+    _id: org._id,
+  };
+  var updates = {
+    $set: {
+      'kms.crn': crn,
+    }
+  };
+  await models.Organization.updateOne(search, updates);
+  return crn;
+};
+
+var rotateKmsKeyForOrg = async({ org, metroConf })=>{
```

Comment:
When is `rotateKmsKeyForOrg` called?  I suspect we need a 'featurespec' describing the expected flow (e.g. I'm used to keys being generated by SREs and stored elsewhere rather than auto-generated and presumably auto-rotated... and not clear where the rotation would occur when there are multiple instances accessing the buckets, etc).